cds_ils/migrator/acquisition/orders.py

```diff
@@ -0,0 +1,225 @@
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS-ILS acquisition order migrator API.
+
+Acquisition Order CDS fields:
+              legacy_id: 1
+         id_crcBORROWER: 2
+                barcode: J00227767
+period_of_interest_from: 2010-08-18 00:00:00
+  period_of_interest_to: 2011-08-18 00:00:00
+          id_crcLIBRARY: 0
+           request_date: 0000-00-00 00:00:00
+          expected_date: 0000-00-00 00:00:00
+           arrival_date: 2010-08-18 00:00:00
+               due_date: 2010-09-17 00:00:00
+            return_date: 0000-00-00 00:00:00
+                 status: returned
+                   cost: 0 EUR
+              item_info: {'publisher': '', 'isbn': '', 'title': '', 'authors': '', 'edition': '', 'place': '', 'year': ''}
+           request_type: acq-book
+      borrower_comments: NULL
+      only_this_edition: None
+          library_notes: {}
+            budget_code:
+  overdue_letter_number: 0
+    overdue_letter_date: 0000-00-00 00:00:00
+
+
+CDS statuses          ILS AcqOrder statuses
+cancelled          -> "CANCELLED"
+new                -> "PENDING"
+on order           -> "ORDERED"
+received           -> "RECEIVED"
+proposal-received  -> "RECEIVED" It was doc request and resolved with acquisition
+
+proposal-new       ->   3 records -> PENDING doc_request
+proposal-on order  ->  13 records -> ? PENDING doc_request for now
+proposal-put aside ->  86 records -> REJECTED with reason in library_notes
+"""
+
+import json
+
+import click
+from flask import current_app
+from invenio_db import db
+
+from cds_ils.migrator.api import (
+    bulk_index_records,
+    import_record,
+    model_provider_by_rectype,
+)
+from cds_ils.migrator.acquisition.vendors import get_vendor_pid_by_legacy_id
+from cds_ils.migrator.errors import AcqOrderError, ItemMigrationError
+from cds_ils.migrator.items.api import get_item_by_barcode
+from cds_ils.migrator.patrons.api import get_user_by_legacy_id
+from cds_ils.migrator.utils import get_acq_ill_notes, get_cost
+
+MIGRATION_DOCUMENT_PID = ""
+DEFAULT_ITEM_MEDIUM = "ELECTRONIC"
+
+
+def get_status(record):
+    """Map CDS status to a valid ILS status."""
+    ILS_ORDER_STATUSES = ["PENDING", "ORDERED", "RECEIVED", "CANCELLED"]
+
+    status = record["status"]
+    if status == "new":
+        return ILS_ORDER_STATUSES[0]
+    if status == "on order":
+        return ILS_ORDER_STATUSES[1]
+    if status == "received":
+        return ILS_ORDER_STATUSES[2]
+    if status == "cancelled":
+        return ILS_ORDER_STATUSES[3]
+    raise AcqOrderError(
+        "This status is ambiguous for acquisition order!\n{}".format(record)
+    )
+
+
+def get_patron_pid(record):
+    """Get patron_pid from existing users."""
+    anonym = current_app.config["ILS_PATRON_ANONYMOUS_CLASS"](-2)
+    patron_pid = anonym.id
+    user = get_user_by_legacy_id(record["id_crcBORROWER"])
+    if user:
+        patron_pid = user.pid
```

Comment:
This is where I got inspired from, https://github.com/CERNDocumentServer/cds-ils/blob/50a8dc3d15b9fdfc5dce998ca9078d7b2e078941/cds_ils/migrator/loans/api.py#L47, and since everything was imported and assigned I didn't even think about it, will check it again.